src/D2L.Bmx/Program.cs

```diff
@@ -233,10 +233,16 @@
 					throw new BmxException( "Failed to initialize BMX directory (~/.bmx)", ex );
 				}
 			}
+			// clean up temp files from earlier runs
+			if( Directory.Exists( BmxPaths.TEMP_DIR ) ) {
+				try {
+					Directory.Delete( BmxPaths.TEMP_DIR, recursive: true );
+				} catch( Exception ) {
+					// ignore errors as this doesn't impact normal BMX usage
+				}
+			}
 
-			UpdateHandler.Cleanup();
```

Comment:
inlining the logic here, so we can see all directory init logic in one place (in this middleware)